### ISSUE TITLE
Test fails around number conversion

### ISSUE THREAD

**Issue by qykth-git** [shirok/Gauche#96]
There is some test error since pull request #84 

My `gosh` version 0.9.5_pre1 on 32 bit x86 machine makes these test error that comes from `test/number.scm`:

```
test exact (greatest-fixnum), expects 536870912 ==> ERROR: GOT 536870911
test round->exact, expects 536870912 ==> ERROR: GOT 536870911
test floor->exact, expects 536870912 ==> ERROR: GOT 536870911
test ceiling->exact, expects 536870912 ==> ERROR: GOT 536870911
test truncate->exact, expects 536870912 ==> ERROR: GOT 536870911
```

Above error was comes from these tests:
```scheme
(test* "exact (greatest-fixnum)" (+ (greatest-fixnum) 1) (exact (inexact (greatest-fixnum))))
(test* "round->exact" (+ (greatest-fixnum) 1) (round->exact (inexact (greatest-fixnum))))
(test* "floor->exact" (+ (greatest-fixnum) 1) (floor->exact (inexact (greatest-fixnum))))
(test* "ceiling->exact" (+ (greatest-fixnum) 1) (ceiling->exact (inexact (greatest-fixnum))))
(test* "truncate->exact" (+ (greatest-fixnum) 1) (truncate->exact (inexact (greatest-fixnum))))
```

And here is in interactive mode:
```scheme
gosh> (greatest-fixnum)
536870911 ;; == #x1fffffff
gosh> (+ (greatest-fixnum) 1)
536870912 ;; == #x20000000
gosh> (inexact (greatest-fixnum))
536870911.0
gosh> (exact (inexact (greatest-fixnum)))
536870911
```

I think those tests were only works on 64 bit machine, but I can't test it.

**Comment by shirok** [comment]
Yup.  I'm aware of it for some time but don't have readily accessible 32bit machines.  I think I can boot old laptop and take a look.

**Comment by shirok** [comment]
Fixed in commit #706ce58

**Comment by qykth-git** [comment]
Thanks. It also works on my machine.
